modules/images/webp-uploads/helper.php

```diff
@@ -65,6 +65,28 @@ function webp_uploads_get_upload_image_mime_transforms() {
  * @return array|WP_Error An array with the file and filesize if the image was created correctly otherwise a WP_Error
  */
 function webp_uploads_generate_additional_image_source( $attachment_id, array $size_data, $mime, $destination_file_name = null ) {
+
+	/**
+	 * Filter to allow the generation of additional image sources, in which a defined mime type
+	 * can be transformed and create additional mime types for the file.
+	 *
+	 * @since n.e.xt
+	 *
+	 * @param array  $image         Image data {'path'=>string, 'file'=>string, 'width'=>int, 'height'=>int, 'mime-type'=>string} or empty array.
+	 * @param int    $attachment_id The ID of the attachment from where this image would be created.
+	 * @param string $size          The size name that would be used to create this image, out of the registered subsizes.
+	 * @param array  $size_data     An array with the dimensions of the image: height, width and crop {'height'=>int, 'width'=>int, 'crop'}.
+	 * @param string $mime          The target mime in which the image should be created.
+	 */
+	$image = (array) apply_filters( 'webp_uploads_pre_generate_additional_image_source', array(), $attachment_id, 'full', $size_data, $mime );
```

Comment:
```suggestion
	$image = (array) apply_filters( 'webp_uploads_pre_generate_additional_image_source', null, $attachment_id, 'full', $size_data, $mime );
```

Reply:
Looks this function uses all the time `full` but that's not the accurate as the size changes depending on the parameters passed to this function.

Reply:
I don't think we should cast to `array`  due the filter can return `WP_Error` as well.

Reply:
@mitogh I have removed cast to `array` now.

About the size default to full for all the sizes, I am confused on how do we get the the size name here as we don't have any size name related parameters in the function. so should we get the size name using `wp_get_registered_image_subsizes` and looping through the sizes and the get the matching size?

Reply:
Good call I think one of the ways we can do it is by adding a new parameter to the top level function with the size name as the caller to `webp_uploads_generate_additional_image_source` has the information about the size name.

Let me know what do you think about that. 

I did a quick search and looks like this one is only used on the `full` size tho but I might be wrong here.